extensions/GitExtension/FileExplorerGitIntegration/Models/StatusCache.cs

```diff
@@ -292,13 +307,50 @@ private void ParseStatus(string? statusString, GitRepositoryStatus repoStatus, s
                 var filePath = Path.Combine(relativeDir, line.Substring(2)).Replace(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
                 repoStatus.Add(filePath, new GitStatusEntry(filePath, FileStatus.NewInWorkdir));
             }
+            else if (line.StartsWith("# branch.oid ", StringComparison.Ordinal))
+            {
+                // For porcelain=v2, the branch status line has the following format:
+                //   # branch.oid <sha>
+                // For now, we only care about the <sha>.
+                repoStatus.Sha = line.Substring(13);
+            }
+            else if (line.StartsWith("# branch.ab ", StringComparison.Ordinal))
+            {
+                // For porcelain=v2, the branch status line has the following format:
+                //   # branch.ab +<ahead> -<behind>
+                // For now, we only care about the <ahead> and <behind>.
+                var pieces = line.Split(' ', 4);
+                var aheadBy = int.Parse(pieces[2][1..], CultureInfo.InvariantCulture);
+                var behindBy = int.Parse(pieces[3][1..], CultureInfo.InvariantCulture);
+                repoStatus.AheadBy = aheadBy;
+                repoStatus.BehindBy = behindBy;
+            }
+            else if (line.StartsWith("# branch.upstream ", StringComparison.Ordinal))
+            {
+                // For porcelain=v2, the branch status line has the following format:
+                //   # branch.upstream <upstream>
+                // For now, we only care about the <upstream>.
+                repoStatus.UpstreamBranch = line.Substring(17);

```

Comment:
Assuming 17 chars feels brittle, when git branches can not contain spaces (see https://git-scm.com/docs/git-check-ref-format They cannot have ASCII control characters (i.e. bytes whose values are lower than \040, or \177 `DEL`), space ` `, tilde `~`, caret `^`, or colon `:` anywhere.). Instead, split on `' '`.